### ISSUE TITLE
Tests that reuse `SANDBOX_DIR`

### ISSUE THREAD

**Issue by mgree** [binpash/try#41]
We should have tests that run multiple commands in the same sandbox. (Not quite the same as `try explore` from #5, but related.)

**Comment by gliargovas** [comment]
It seems like reusing overlay directories does not currently work:
```ShellSession
$ mkdir foo
$ try -D foo "echo foo > file.txt"
Warning: Failed mounting /proj as an overlay, see /tmp/tmp.O75mMQXBe0
Warning: Failed mounting /sys/kernel/tracing as an overlay, see /tmp/tmp.O75mMQXBe0
Warning: Failed mounting /proj/pash-PG0 as an overlay, see /tmp/tmp.O75mMQXBe0
$ try -D foo "wc -l file.txt > lines.txt"
mkdir: cannot create directory ‘foo/upperdir//bin’: File exists
mkdir: cannot create directory ‘foo/workdir//bin’: File exists
mkdir: cannot create directory ‘foo/temproot//bin’: File exists
mkdir: cannot create directory ‘foo/upperdir//boot’: File exists
mkdir: cannot create directory ‘foo/workdir//boot’: File exists
mkdir: cannot create directory ‘foo/temproot//boot’: File exists
mkdir: cannot create directory ‘foo/upperdir//dev’: File exists
mkdir: cannot create directory ‘foo/workdir//dev’: File exists
mkdir: cannot create directory ‘foo/temproot//dev’: File exists
mkdir: cannot create directory ‘foo/upperdir//etc’: File exists
mkdir: cannot create directory ‘foo/workdir//etc’: File exists
mkdir: cannot create directory ‘foo/temproot//etc’: File exists
mkdir: cannot create directory ‘foo/upperdir//home’: File exists
mkdir: cannot create directory ‘foo/workdir//home’: File exists
mkdir: cannot create directory ‘foo/temproot//home’: File exists
mkdir: cannot create directory ‘foo/upperdir//lib’: File exists
mkdir: cannot create directory ‘foo/workdir//lib’: File exists
mkdir: cannot create directory ‘foo/temproot//lib’: File exists
mkdir: cannot create directory ‘foo/upperdir//lib32’: File exists
mkdir: cannot create directory ‘foo/workdir//lib32’: File exists
mkdir: cannot create directory ‘foo/temproot//lib32’: File exists
mkdir: cannot create directory ‘foo/upperdir//lib64’: File exists
mkdir: cannot create directory ‘foo/workdir//lib64’: File exists
mkdir: cannot create directory ‘foo/temproot//lib64’: File exists
mkdir: cannot create directory ‘foo/upperdir//libx32’: File exists
mkdir: cannot create directory ‘foo/workdir//libx32’: File exists
mkdir: cannot create directory ‘foo/temproot//libx32’: File exists
mkdir: cannot create directory ‘foo/upperdir//local’: File exists
mkdir: cannot create directory ‘foo/workdir//local’: File exists
mkdir: cannot create directory ‘foo/temproot//local’: File exists
mkdir: cannot create directory ‘foo/upperdir//lost+found’: File exists
mkdir: cannot create directory ‘foo/workdir//lost+found’: File exists
mkdir: cannot create directory ‘foo/temproot//lost+found’: File exists
mkdir: cannot create directory ‘foo/upperdir//media’: File exists
mkdir: cannot create directory ‘foo/workdir//media’: File exists
mkdir: cannot create directory ‘foo/temproot//media’: File exists
mkdir: cannot create directory ‘foo/upperdir//mnt’: File exists
mkdir: cannot create directory ‘foo/workdir//mnt’: File exists
mkdir: cannot create directory ‘foo/temproot//mnt’: File exists
mkdir: cannot create directory ‘foo/upperdir//opt’: File exists
mkdir: cannot create directory ‘foo/workdir//opt’: File exists
mkdir: cannot create directory ‘foo/temproot//opt’: File exists
mkdir: cannot create directory ‘foo/upperdir//proc’: File exists
mkdir: cannot create directory ‘foo/workdir//proc’: File exists
mkdir: cannot create directory ‘foo/temproot//proc’: File exists
mkdir: cannot create directory ‘foo/upperdir//proj’: File exists
mkdir: cannot create directory ‘foo/workdir//proj’: File exists
mkdir: cannot create directory ‘foo/temproot//proj’: File exists
mkdir: cannot create directory ‘foo/upperdir//root’: File exists
mkdir: cannot create directory ‘foo/workdir//root’: File exists
mkdir: cannot create directory ‘foo/temproot//root’: File exists
mkdir: cannot create directory ‘foo/upperdir//run’: File exists
mkdir: cannot create directory ‘foo/workdir//run’: File exists
mkdir: cannot create directory ‘foo/temproot//run’: File exists
mkdir: cannot create directory ‘foo/upperdir//sbin’: File exists
mkdir: cannot create directory ‘foo/workdir//sbin’: File exists
mkdir: cannot create directory ‘foo/temproot//sbin’: File exists
mkdir: cannot create directory ‘foo/upperdir//share’: File exists
mkdir: cannot create directory ‘foo/workdir//share’: File exists
mkdir: cannot create directory ‘foo/temproot//share’: File exists
mkdir: cannot create directory ‘foo/upperdir//srv’: File exists
mkdir: cannot create directory ‘foo/workdir//srv’: File exists
mkdir: cannot create directory ‘foo/temproot//srv’: File exists
mkdir: cannot create directory ‘foo/upperdir//sys’: File exists
mkdir: cannot create directory ‘foo/workdir//sys’: File exists
mkdir: cannot create directory ‘foo/temproot//sys’: File exists
mkdir: cannot create directory ‘foo/upperdir//tmp’: File exists
mkdir: cannot create directory ‘foo/workdir//tmp’: File exists
mkdir: cannot create directory ‘foo/temproot//tmp’: File exists
mkdir: cannot create directory ‘foo/upperdir//users’: File exists
mkdir: cannot create directory ‘foo/workdir//users’: File exists
mkdir: cannot create directory ‘foo/temproot//users’: File exists
mkdir: cannot create directory ‘foo/upperdir//usr’: File exists
mkdir: cannot create directory ‘foo/workdir//usr’: File exists
mkdir: cannot create directory ‘foo/temproot//usr’: File exists
mkdir: cannot create directory ‘foo/upperdir//var’: File exists
mkdir: cannot create directory ‘foo/workdir//var’: File exists
mkdir: cannot create directory ‘foo/temproot//var’: File exists
Warning: Failed mounting /proj as an overlay, see /tmp/tmp.IvI1nbUjSz
Warning: Failed mounting /sys/kernel/tracing as an overlay, see /tmp/tmp.IvI1nbUjSz
Warning: Failed mounting /proj/pash-PG0 as an overlay, see /tmp/tmp.IvI1nbUjSz
```

**Comment by gliargovas** [comment]
This issue can be solved by using `mkdir -p`. Not sure yet about other side effects of this.

**Comment by ericzty** [comment]
On Sun Jun 25, 2023 at 11:57 AM EDT, Georgios Liargkovas wrote:
> This issue can be solved by using `mkdir -p`. Not sure yet about other side effects of this.

I think that's a good idea.
+1

-- 
Eric

**Comment by mgree** [comment]
I think it's better to detect that it's an existing try result; if there are other, extra files (eg, a file in lowerdir) we should warn or error out.

**Comment by angelhof** [comment]
This seems like a bug to me and not an enhancement so I think we need to fix it before 0.2
